boilerplate/openshift/golang-osd-operator/csv-generate/csv-generate.sh

```diff
@@ -79,14 +79,66 @@ if [ "$diff_generate" = true ] ; then
 else
     rm -rf "$SAAS_OPERATOR_DIR"
     git clone --branch "$operator_channel" ${GIT_PATH} "$SAAS_OPERATOR_DIR"
+
+
+    # PATH to saas file in app-interface
+    SAAS_FILE_URL="https://gitlab.cee.redhat.com/service/app-interface/raw/master/data/services/osd-operators/cicd/saas/saas-${operator_name}.yaml"
+
+    # MANAGED_RESOURCE_TYPE
+    # SAAS files contain the type of resources managed within the OC templates that
+    # are being applied to hive.
+    # For on cluster resources this should always be of type "SelectorSyncSet" resources otherwise
+    # wouldn't by sync'd to cluster. So we're explicity selecting the first element in the array.
+    # We can safely assume anything that is not of type "SelectorSyncSet" is being applied to hive only
+    # From this we'll assume that the namespace reference in resourceTemplates to be:
+    # For customer clusters: /services/osd-operators/namespace/<hive shard>/namespaces/cluster-scope.yaml
+    # For hive clusters: /services/osd-operators/namespace/<hive shard>/namespaces/<namespace name>.yaml
+    MANAGED_RESOURCE_TYPE=$(curl -s "${SAAS_FILE_URL}" | \
+            docker run --rm -i quay.io/app-sre/yq:3.4.1 yq r - "managedResourceTypes[0]"
+    )
+
+    if [[ "${MANAGED_RESOURCE_TYPE}" == "" ]]; then
+        echo "Unabled to determine if SAAS file managed resource type"
+        exit 1
+    fi
+
+    # Output our result and make it easier to read by defining a Hive bool
+    HIVE=false
+    if [[ "${MANAGED_RESOURCE_TYPE}" == "SelectorSyncSet" ]]; then
+        echo "SAAS file is NOT applied to Hive, MANAGED_RESOURCE_TYPE=$MANAGED_RESOURCE_TYPE"
+        HIVE=true
```

Comment:
```suggestion
        HIVE=false
```
?

Reply:
redundant now.

Reply:
> This looks workable except for the mistake noted inline
> 
> but
> 
> Perhaps it would be easier to require the consumer to set a Makefile variable indicating whether they're hive or not. The less we can rely on specific yaml formats (or even graphql queries) the better.

I don't disagree with this. We have protection in that if we guess wrong the hash should be blank and we'll error. In the interest of time I think lets move forward with this (since its only one merge and we don't have to touch the other operators) and fix this later. https://issues.redhat.com/browse/OSD-6995